assets/model_monitoring/components/src/action_analyzer/contracts/action.py

```diff
@@ -0,0 +1,152 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+"""Action Class."""
+
+import datetime
+import uuid
+import json
+from enum import Enum
+from abc import ABC, abstractmethod
+from action_analyzer.utils.utils import convert_to_camel_case
+
+class ActionSample:
+    """Action sample class."""
+
+    def __init__(self,
+                 question: str,
+                 answer: str,
+                 lookup_score: float,
+                 debugging_info: str,
+                 retrieval_query_type: str,
+                 retrieval_top_k: int,
+                 prompt_flow_input: str) -> None:
+        """Create an action sample.
+
+        Args:
+            question(str): the input question of the flow.
+            answer(str): the output answer of the flow
+            lookup_score(float): the retrieval document look up score.
+            debugging_info(str): the json string of debugging info in a span tree structure.
+            retrieval_query_type(str): the retrieval query type in the retrieval span.
+            retrieval_top_k(int): the retrieval top k value in the retrieval span.
+            prompt_flow_input(str): the json str of prompt flow input.
+        """
+        self.question = question
+        self.answer = answer
+        self.lookup_score = lookup_score
+        self.debugging_info = debugging_info
+        self.retrieval_query_type = retrieval_query_type
+        self.retrieval_top_k = retrieval_top_k
+        self.prompt_flow_input = prompt_flow_input
+
+
+    def to_json_str(self) -> str:
+        """Convert an action sample object to json string."""
+        attribute_dict =  self.__dict__
+        json_out = {}
+        for key, val in attribute_dict.items():
+            json_out[convert_to_camel_case(key)] = val
+
+        return json.dumps(json_out)
+
+
+class ActionType(Enum):
+    """Action type."""
+
+    METRICS_VIOLATION_INDEX_ACTION = 1
+    BAD_RETRIEVAL_SCORE_INDEX_ACTION = 2
+
+
+class Action():
+    """Action class."""
+
+    def __init__(self,
+                 action_type: ActionType, 
+                 description: str,
+                 confidence_score: float,
+                 query_intention: str,
+                 deployment_id: str,
+                 run_id: str,
+                 positive_samples: list[str],
```

Comment:
what is the downside if we do list[ActionSample] ?

Reply:
changed